lib/fault_handler/fault_handler.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <arch/cpu.h>
+#include <logging/log_ctrl.h>
+#include <logging/log.h>
+#include <fatal.h>
+
+LOG_MODULE_REGISTER(fault_handler, CONFIG_FAULT_HANDLER_LOG_LEVEL);
+
+__weak void system_stop(unsigned int reason)
+{
+	ARG_UNUSED(reason);
+
+	(void)arch_irq_lock();
+	for (;;) {
+		/* Spin endlessly */
+	}
+}
+
+__weak void system_reset(unsigned int reason)
+{
+	ARG_UNUSED(reason);
+
+	NVIC_SystemReset();
+}
+
+void k_sys_fatal_error_handler(unsigned int reason,
+			       const z_arch_esf_t *esf)
+{
+	ARG_UNUSED(esf);
+
+	LOG_PANIC();
+
+	if (IS_ENABLED(CONFIG_DEBUG)) {
+		LOG_ERR("Halting system");
+		system_stop(reason);
```

Comment:
If we are going to do the same thing as the default "k_sys_fatal_error_handler", why even bother overriding it when CONFIG_DEBUG ?

Reply:
I can imagine a user would want to enable DEBUG and yet still have a resetting system.

I would suggest we either don't have CONFIG_DEBUG affect the fault handler at all, or have it affect the default value (so that it can still be set with or without CONFIG_DEBUG).

Reply:
Maybe it was some overkill to have also system halt here. So I removed it and add dependencies to default value of the `RESET_ON_FAULT`